core/include/GenJet.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "Particle.h"
+
+
+class GenJet : public Particle{
```

Comment:
Do we want to inherit from `FlavorParticle` instead for parity with the `Jet` class? I realise though that it isn't perfect, since like in the `Jet` class you'd have to add in a separate getter/setter for hadron flavour.